### PR TITLE
Move CXShim and CXTest to a separate repo

### LINKED ISSUE #117: Build failure when use CXShim + Combine + iOS (No such module 'Combine')
When use CXShim with system Combine and target on 'Any iOS Device', we got "No such module 'Combine'" error. As reported in #116 and sbertix/Swiftagram#191.

This is because `CXShim` always have low system requirement. To re-export system Combine, we need to bump platform to iOS 13.0.

However, system requirement is applied to both `CombineX` and `CXShim`. We can make it configurable via environment variable, but we can't make them both work out of the box. Currently `CX_CONTINUOUS_INTEGRATION=YES CX_COMBINE_IMPLEMENTATION=Combine` will force high platform and `CX_COMBINE_IMPLEMENTATION=CombineX` will force low platform.

I believe `CombineX` and `CXShim` are equally important. Mandatory env configuration can make integration significantly more obscure. The best solution I can think of is to move `CXShim` to a separate repo. So we can lift system requirement just for `CXShim`. This would be a breaking change.

### PR DESCRIPTION
**Pull request by ddddxxx**
Use git submodule to avoid code duplicate as we still need `CXTest` for test targets.

resolve #117

### CODE REVIEW

**Comment by ddddxxx** [comment]
@luoxiu I'd like to move `CXCompatible` to CXShim repo, what do you think?

**Comment by ddddxxx** [comment]
@luoxiu would you please review this pr?

File: CombineX.podspec [hunk 1]
```diff
@@ -15,27 +15,26 @@ Pod::Spec.new do |s|
...
-
   s.subspec "Main" do |ss|
     ss.source_files = "Sources/CombineX/**/*.swift"
-    ss.dependency "CombineX/CXLibc"
     ss.dependency "CombineX/CXUtility"
+<<<<<<< HEAD
+    # ss.dependency "Runtime"
+=======
     ss.dependency "CombineX/CXNamespace"
+>>>>>>> master
```
Comment (luoxiu) [inline CombineX.podspec:29]: Unresolved conflicts here.

Hum, don't worry, I will fix this before the next release!

**Review by luoxiu — APPROVED** [review]
As far as I can see, there are only structural changes, thanks for the hard work!

And yes, I totally agree with the idea that CXShim and CXCompatible should be put together in a new separate project!

**Pull request merged**